internal/agent/config.go

```diff
@@ -99,14 +103,15 @@ func NewDefault() *Config {
 		reader:               fileio.NewReader(),
 		LogLevel:             logrus.InfoLevel.String(),
 	}
-}
 
-func (cfg *Config) SetTestRootDir(rootDir string) {
-	klog.Warning("Setting testRootDir is intended for testing only. Do not use in production.")
-	if cfg.reader != nil {
-		cfg.reader.SetRootdir(rootDir)
+	if value := os.Getenv(TestRootDirEnvKey); value != "" {
+		klog.Warning("Setting testRootDir is intended for testing only. Do not use in production.")
+		c.reader.SetRootdir(value)
+		c.testRootDir = value
```

Comment:
Do we need to mutate the config in the agent vs cmd? Ideally, the config is complete and validated before here.

Reply:
The challenged I faced is `Validate()` currently tests whether specified files exist and are readable, so the testRootDir needs to be set before it.